portfolio/src/main/webapp/script.js

```diff
@@ -199,16 +244,69 @@ async function deleteMessage(msg) {
   });
 }
 
-function openReplies(msgID, tag) { 
-  document.getElementById("blogcontent").style.display = "none";
-  document.getElementById("blogreplycontent").style.display = "block";
-  document.getElementById('parentID').value = msgID;
-  document.getElementById('tags-reply').value = tag;
+// Displays the reply form under a post when its reply button is clicked.
+// Removes the reply form if button is clicked again, and so on...
+function handleReplies(msgID, msgSender, tag, postEleID) { 
+  // The form appears when a reply button is clicked an even number of times, so
+  // If a users clicks a different reply button, set counter to zero for that button
+  if (clickedDifferentReplyButton(msgID)) {
+    replyClickCounter = 0;
+  }
+  
+  // Remove the form when a button is clicked an odd number of times in a row.
+  if (replyClickCounter % 2 != 0) {
```

Comment:
I think a cleaner way to do this is check if the element is hidden or not.That would more directly relate to what the code is trying to accomplish. I think you can check what the current style.display is set to.

Reply:
I think that way if a reply button is shown for some post and user clicks the reply button of another post, the reply box will be hidden.(because it's displayed somewhere), which is what was happening but I didn't like it. But in this case we want to show the box (no matter the box's display status) since the user is trying to reply to some other post. That's why the previous condition checks if the button clicked is for a different post. 
I don't know if that makes any difference.